src/slack.ts

```diff
@@ -1,25 +1,55 @@
 import { IncomingWebhook } from "@slack/webhook";
-import { SLACK_WEBHOOK_URL, SLACK_DAY_INTERVAL } from "./config";
+import { CommitWithDiagnoses } from "./db";
+import { SLACK_WEBHOOK_URL } from "./config";
+import { SectionBlock, DividerBlock } from "@slack/types";
 
 interface SendSlackSummaryParams {
-  numCommits: number;
-  numCommitDiagnoses: number;
+  topCommits: CommitWithDiagnoses[];
 }
 const webhook = new IncomingWebhook(SLACK_WEBHOOK_URL);
 
+const dividerBlock: DividerBlock = {
+  type: "divider",
+};
+
+function createMarkdownSection(text: string): SectionBlock {
+  return {
+    type: "section",
+    text: {
+      type: "mrkdwn",
+      text,
+    },
+  };
+}
+
+function createTopCommitsBlock(topCommits: CommitWithDiagnoses[]) {
+  const header =
+    "*Top 10 commit messages with highest lint violation scores:*\n";
+  const messageBody = topCommits
+    .map((commit) => {
+      const paddedScore = String(commit.score).padEnd(3, " ");
+      const shortId = commit.id.slice(0, 8);
+
+      const messageComponents: string[] = [
+        `• Score: ${paddedScore} | ${commit.repo_name}: <${commit.url}|${shortId}>`,
```

Comment:
- let's use the `org/repo@commit` syntax as used by [GitHub](https://help.github.com/en/github/writing-on-github/autolinked-references-and-urls#commit-shas)
- we can include the commit message (subject line) in the summary